src/format/parsed.rs

```diff
@@ -553,7 +553,16 @@ impl Parsed {
     /// Either way those fields have to be consistent to each other.
     pub fn to_naive_datetime_with_offset(&self, offset: i32) -> ParseResult<NaiveDateTime> {
         let date = self.to_naive_date();
-        let time = self.to_naive_time();
+        let time = if self.hour_div_12.is_none()
```

Comment:
What's your consideration for doing this here? Why not do this inside `to_naive_time()`?